packages/dashboard/src/Dashboard.tsx

```diff
@@ -70,31 +78,82 @@ function Dashboard() {
         } else if (isDebugMessage(message)) {
           const { payload } = message;
           console.log(payload.message);
-          respond({ id: message.id }, connectedSocket);
+          respond({ id: message.id }, socket);
         }
       }
     );
+    socket.send("ready");
+
+    setSubscribeSocket(socket);
+  };
+
+  const initializePubSocket = async () => {
+    if (publishSocket && publishSocket.readyState === WebSocket.OPEN) return;
 
-    connectedSocket.send("ready");
+    const { publishPort } = await getPorts();
+    const socket = await initializeSocket(
+      publishPort,
+      (event: WebSocket.MessageEvent) => {
+        if (typeof event.data !== "string") {
```

Comment:
The publish socket probably doesn't need any event handler at this point. We can use the socket to make a request for data but we don't need every message.